pkg/scheduler/objects/queue.go

```diff
@@ -74,42 +74,46 @@ type Queue struct {
 	// The queue properties should be treated as immutable the value is a merge of the
 	// parent properties with the config for this queue only manipulated during creation
 	// of the queue or via a queue configuration update.
-	properties             map[string]string
-	adminACL               security.ACL        // admin ACL
-	submitACL              security.ACL        // submit ACL
-	maxResource            *resources.Resource // When not set, max = nil
-	guaranteedResource     *resources.Resource // When not set, Guaranteed == 0
-	isLeaf                 bool                // this is a leaf queue or not (i.e. parent)
-	isManaged              bool                // queue is part of the config, not auto created
-	stateMachine           *fsm.FSM            // the state of the queue for scheduling
-	stateTime              time.Time           // last time the state was updated (needed for cleanup)
-	maxRunningApps         uint64
-	runningApps            uint64
-	allocatingAcceptedApps map[string]bool
-	template               *template.Template
-	queueEvents            *schedEvt.QueueEvents
+	properties                 map[string]string
+	adminACL                   security.ACL        // admin ACL
+	submitACL                  security.ACL        // submit ACL
+	maxResource                *resources.Resource // When not set, max = nil
+	guaranteedResource         *resources.Resource // When not set, Guaranteed == 0
+	isLeaf                     bool                // this is a leaf queue or not (i.e. parent)
+	isManaged                  bool                // queue is part of the config, not auto created
+	stateMachine               *fsm.FSM            // the state of the queue for scheduling
+	stateTime                  time.Time           // last time the state was updated (needed for cleanup)
+	maxRunningApps             uint64
+	runningApps                uint64
+	allocatingAcceptedApps     map[string]bool
+	template                   *template.Template
+	queueEvents                *schedEvt.QueueEvents
+	quotaChangePreemptionDelay uint64
+	quotaChangePreemptionTimer *time.Timer
```

Comment:
The curical question of this PR: do we want to initiate the preemption logic from a Timer goroutine? Because I'm very much against that. I think it's much easier to reason about the quota preemption vs other preemption if it happens in the scheduling cycle.